turbopack/crates/turbo-tasks/src/registry.rs

```diff
@@ -5,21 +5,16 @@ use once_cell::sync::Lazy;
 
 use crate::{
     FxDashMap, TraitType, ValueType,
-    id::{FunctionId, TraitTypeId, ValueTypeId},
+    id::{TraitTypeId, ValueTypeId},
     id_factory::IdFactory,
     native_function::NativeFunction,
     no_move_vec::NoMoveVec,
 };
 
-static FUNCTION_ID_FACTORY: IdFactory<FunctionId> = IdFactory::new_const(
-    FunctionId::MIN.to_non_zero_u64(),
-    FunctionId::MAX.to_non_zero_u64(),
-);
-static FUNCTIONS_BY_NAME: Lazy<FxDashMap<&'static str, FunctionId>> = Lazy::new(FxDashMap::default);
-static FUNCTIONS_BY_VALUE: Lazy<FxDashMap<&'static NativeFunction, FunctionId>> =
+static NAME_TO_FUNCTION: Lazy<FxDashMap<&'static str, &'static NativeFunction>> =
```

Comment:
do you think it would make sense to swtich to a `RwLock<HashMap...>>`  there are no concurrent updates so i don't think sharding is buying us anything

Reply:
yes that makes sense. Also switch to a FxHashMap.